src/App.tsx

```diff
@@ -1,24 +1,274 @@
+    return todos.every(todo => todo.completed);
+  }, [todos]);
+
+  const [isEdit, setIsEdit] = useState(false);
+
+  const getTodos = async () => {
+    try {
+      const todosFromServer = await callGetTodos(USER_ID);
+
+      setTodos(todosFromServer);
+      setHasError(false);
+    } catch (error) {
+      setHasError(true);
+      if (error instanceof Error) {
+        setErrorType(ErrorType.LOAD);
+      }
+    }
+  };
+
+  useEffect(() => {
+    getTodos();
+  }, []);
+
+  const addTodo = useCallback(async (titleOfTodo: string) => {
+    const preparedTitle = titleOfTodo.trim();
+
+    if (!preparedTitle) {
+      setErrorType(ErrorType.TITLE);
+      setHasError(true);
+      setTitle('');
+
+      return;
+    }
+
+    const newTodo: RequestTodo = {
+      userId: USER_ID,
+      title: preparedTitle,
+      completed: false,
+    };
+
+    try {
+      setTempTodo({ ...newTodo, id: 0 });
+      await callAddTodo(newTodo);
+      await getTodos();
+      setTempTodo(null);
+      setTitle('');
+      setHasError(false);
+    } catch (error) {
+      setHasError(true);
+      setErrorType(ErrorType.ADD);
+    } finally {
+      setTempTodo(null);
+    }
+  }, []);
+
+  const deleteTodo = useCallback(async (id: number) => {
+    setTodoWithLoader(id);
+
+    try {
+      await callDeleteTodo(id);
+      await getTodos();
+    } catch (error) {
+      setErrorType(ErrorType.DELETE);
+      setHasError(true);
+    } finally {
+      setTodoWithLoader(0);
+    }
+  }, []);
+
+  const deleteCompleted = async (id: number) => {
+    setClearCompleted(true);
+
+    try {
+      await deleteTodo(id);
+      setTodos((currentTodos) => currentTodos.filter(todo => todo.id !== id));
+    } catch {
+      setErrorType(ErrorType.DELETE);
+    } finally {
+      setClearCompleted(false);
+    }
+  };
+
+  const editTodoStatus = useCallback(async (editedTodo: Todo) => {
+    setTodoWithLoader(editedTodo.id);
+
+    const undatedTodo = {
+      ...editedTodo,
+      completed: !editedTodo.completed,
+    };
+
+    try {
+      await callEditTodo(undatedTodo);
+      await getTodos();
+    } catch {
+      setErrorType(ErrorType.UPDATE);
+      setHasError(true);
+    } finally {
+      setTodoWithLoader(0);
+    }
+  }, []);
+
+  const editTodoTitle = useCallback(async (editedTodo: Todo,
+    newTitle: string) => {
+    setIsEdit(true);
+    if (newTitle === editedTodo.title) {
+      setEditingId(0);
+
+      return;
+    }
+
+    setTodoWithLoader(editedTodo.id);
+    const undatedTodo = { ...editedTodo, title: newTitle };
+
+    try {
+      await callEditTodo(undatedTodo);
+      await getTodos();
+    } catch {
+      setErrorType(ErrorType.UPDATE);
+      setHasError(true);
+    } finally {
+      setTodoWithLoader(0);
+      setEditingId(0);
+      setIsEdit(false);
+    }
+  }, []);
+
+  const handleTitleTodo = useCallback((newTitle: string) => {
+    setTitle(newTitle);
+  }, [title]);
+
+  const handleToggleStatus = async () => {
```

Comment:
you've created so many functions in one component
can we separate them somehow?
move to some child components instead of declare all of them at the parent one?